mentor-match-app/src/components/profile/LoggedUserProfile.jsx

```diff
@@ -0,0 +1,490 @@
+      }
+      fetchFormAnswers()
+    }
+  }, [openDialog, user.uid])
+
+  const initialValues = {
+    name: user.displayName,
+    description: user.profileDescription,
+    affiliation: user.affiliation,
+    title: user.title,
+    location: user.location,
+    profilePicture: user.profilePicture,
+    profileDescription: user.profileDescription,
+    websiteUrl: user.websiteUrl
+  }
+
+  return (
+    <Dialog
+      open={openDialog}
+      onClose={() => setOpenDialog(false)}
+      maxWidth={'lg'}
+      fullWidth
+    >
+      <DialogContent>
+        {loading ? (
+          <Box
+            display="flex"
+            justifyContent="center"
+            alignItems="center"
+            pt={2}
+          >
+            <CircularProgress />
+          </Box>
+        ) : (
+          <Stack
+            //alignItems={'center'}
+            sx={{ width: '100%' }}
+            spacing={1}
+            pb={5}
+          >
+            <IconButton
+              onClick={() => {
+                setOpenDialog(false)
+                setEnableEdit(false)
+              }}
+              color="text.secondary"
+              sx={{ alignSelf: 'flex-end' }}
+            >
+              <CloseIcon />
+            </IconButton>
+            <Stack direction="row" spacing={2} width={'100%'} pr={6} pl={2}>
+              <Formik
+                initialValues={initialValues}
+                validationSchema={schema}
+                onSubmit={onSubmit}
+                sx={{ width: '100%' }}
+              >
+                {({ isSubmitting, values, setFieldValue }) => (
+                  <>
+                    <Stack
+                      width={'35%'}
+                      alignItems={'center'}
+                      justifyContent={'center'}
+                    >
+                      <Box>
+                        <ProfilePicture
+                          img={
+                            enableEdit
+                              ? values.profilePicture
+                              : user.profilePicture
+                          }
+                          size={200}
+                          borderRadius={100}
+                          props={{
+                            position: 'absolute',
+                            left: '8rem',
+                            top: '8rem'
+                          }}
+                        />
+                        {enableEdit && (
+                          <>
+                            <input
+                              type="file"
+                              accept="image/*"
+                              id="upload-profile-picture"
+                              style={{ display: 'none' }}
+                              onChange={async (e) => {
+                                const file = e.target.files[0]
+                                if (file) {
+                                  try {
+                                    // Upload the file to Firebase Storage
```

Comment:
Enhance File Upload Security and Error Handling
```suggestion
                                    // Upload the file to Firebase Storage
- const storageRef = ref(storage, `profilePictures/${file.name}`);
+ const storageRef = ref(storage, `profilePictures/${user.uid}_${Date.now()}_${file.name}`);
```